plasmapy/particles/particle_class.py

```diff
@@ -466,6 +466,110 @@ def __str__(self) -> str:
         """Return the particle's symbol."""
         return self.particle
 
+    def get_json(self) -> dict:
+        """
+        Returns a dictionary object representing the
+        Particle's data
+
+        Examples
+        --------
+
+        >>> import pprint
+        >>> lead = Particle('lead')
+        >>> pprint.pprint(lead.get_json()) #pretty-printed for ease of viewing
+        {'atomic_number': 82,
+         'categories': {'post-transition metal', 'element'},
+         'element': 'Pb',
+         'element_name': 'lead',
+         'ionic_symbol': None,
+         'isotope': None,
+         'periodic_table_entry': {'block': 'p',
+                                 'category': 'post-transition metal',
+                                 'group': 14,
+                                 'period': 6},
+         'symbol': 'Pb'}
+        >>> electron = Particle('e-')
+        >>> pprint.pprint(electron.get_json()) #pretty-printed for ease of viewing
+        {'antiparticle': Particle("e+"),
+         'categories': {'lepton', 'fermion', 'matter', 'electron', 'charged', 'stable'},
+         'element': None,
+         'integer_charge': -1,
+         'ionic_symbol': None,
+         'isotope': None,
+         'roman_symbol': None,
+         'spin': 0.5,
+         'symbol': 'e-'}
```

Comment:
Perhaps we don't need all of this information here? What I had in mind for the issue was something like this workflow:

1. I need to store simulation data that contains information on the `particle`s in that simulation
2. For each particle type, I dump its data to some machine-readable form of text, that could be a str(dictionary) (JSON);
3. I dump the stringified particle data alongside my simulation output, say, in an [`attrs` dictionary of an xarray](http://xarray.pydata.org/en/stable/generated/xarray.DataArray.attrs.html)
4. I load the simulation output from disk, grabbing the string data
5. I turn it back into a dict
6. I use some function to turn that dict back into a `Particle`

So the `dict` does not actually need to contain all the information a `Particle` carries with it; it just needs as much as is necessary to restore the original `Particle` object.

Looking at http://docs.plasmapy.org/en/latest/particles/index.html#classes, this means that for `Particle`s we'd probably just need the `symbol`. It gets a bit trickier with `DimensionlessParticle`s, which will need to store their two numbers, and a bit more trickier with `CustomParticle`s, which will also need their units [(shouldn't be too complex if using existing Astropy functionality)](https://docs.astropy.org/en/stable/units/format.html#string-representations-of-units).

So I think a good direction here would be to make `get_json()` as minimal as possible, then try to implement `parse_json()` that would create the appropriate Particle class from the string produced by `get_json()`. It would also make the docstrings cleaner and testing much simpler.

What do you think?

I'm also sorry I probably misspecified the original issue! Stuff feels obvious when you're typing it down into the void.

Reply:
> So the dict does not actually need to contain all the information a Particle carries with it; it just needs as much as is necessary to restore the original Particle object.

> Looking at http://docs.plasmapy.org/en/latest/particles/index.html#classes, this means that for Particles we'd probably just need the symbol. It gets a bit trickier with DimensionlessParticles, which will need to store their two numbers, and a bit more trickier with CustomParticles, which will also need their units (shouldn't be too complex if using existing Astropy functionality).

Now that I understand the requirement, this clears up a lot of doubts xD. I agree that since the purpose is to store the particles and recover them later, storing the bare minimum is definitely a good way forward as opposed to dumping the entire data regarding the particle. 
Will this report be read only by a machine? If so, then we can probably work only with the bare minimum, however, if there may be a human who will peer at the report at some point, it might be a good idea to have a few additional bits of information included in the JSON (though we can only use the bare minimum to transform the dict back into its respective object).

> I turn it back into a dict

There is a library that does this : https://docs.python.org/3/library/ast.html#ast.literal_eval 
This will come at a cost of having an additional dependency which is not always a good thing. Moreover, parsing the string to a dictionary manually is fairly straightforward.

> I'm also sorry I probably misspecified the original issue! Stuff feels obvious when you're typing it down into the void.

No problem at all :)